firefox-ios/Client/Frontend/Browser/MainMenu/Views/MainMenuViewController.swift

```diff
@@ -223,6 +241,17 @@ class MainMenuViewController: UIViewController,
     func newState(state: MainMenuState) {
         menuState = state
 
+        if let accountData = menuState.accountData {
+            if let iconURL = accountData.iconURL {
+                GeneralizedImageFetcher().getImageFor(url: iconURL) { [weak self] image in
```

Comment:
is this async and then updating UI on a background thread? Not sure,  just pointing out that we should do UI updating on the main thread.

Reply:
It is happening on the main thread.